In [1]:
import pandas as pd


### We need to open the csv with an encoding of ***latin1*** else we'll face an error.

In [27]:
df=pd.read_csv("/content/YCombinatorCompanies.csv", encoding="latin1")
df.head()

Name  ...                                       Description
0       Awesound  ...                              AdWords for podcasts
1   BillionToOne  ...                    Fetal genetic testing in India
2       Bulletin  ...                                 Airbnb for retail
3        Cerplus  ...               B2B marketplace for surplus produce
4  ConnectHealth  ...  Telemedicine and prescription delivery in Africa

[5 rows x 5 columns]

# Column 1: Name cleaning

In [3]:
unique_name=len(df.Name.unique())
all_names= len(df.Name)
print(all_names) , print(unique_name)

962
957


(None, None)

### we have 5 names that are being repeated, we need to check them
1. See if they're duplicted
2. If they're 2 startups with the same name
3. If they updated one of their status

In [6]:
df[df.duplicated(subset=['Name']) == True ]

Name  ...                                        Description
124     Restocks  ...  Push notifications for superconsumers. We make...
132     Shypmate  ...  Shypmate enables a fast, reliable and cost eff...
138  StrongIntro  ...  StrongIntro helps recruiters turn their entire...
338       Supply  ...                    Wireless power, through the air
461        Weave  ...  Better phone service for your dental practice ...

[5 rows x 5 columns]



  ## Restocks

1.  Class difference :  F1 vs W16
2.   Description 




In [8]:
df[df["Name"]== "Restocks"]

Name  ...                                        Description
40   Restocks  ...  Push notifications when sold out products restock
124  Restocks  ...  Push notifications for superconsumers. We make...

[2 rows x 5 columns]



##  Shypmate


1.  Class difference :  F1 vs W16
2.  Description 





In [9]:
df[df["Name"]== "Shypmate"]

Name  ...                                        Description
41   Shypmate  ...  Affordable and fast international peer to peer...
132  Shypmate  ...  Shypmate enables a fast, reliable and cost eff...

[2 rows x 5 columns]



##   StrongIntro
1.  Class difference :  F1 vs W16
2.   Description 
3. Status : Nan vs Dead


In [10]:
df[df["Name"]== "StrongIntro"]

Name  ...                                        Description
43   StrongIntro  ...  Recruiting tool and service that 10xÕs your em...
138  StrongIntro  ...  StrongIntro helps recruiters turn their entire...

[2 rows x 5 columns]



##   Supply
1.  Class difference :  S15 vs W15



In [11]:
df[df["Name"]== "Supply"]

Name  ...                      Description
236  Supply  ...  Wireless power, through the air
338  Supply  ...  Wireless power, through the air

[2 rows x 5 columns]



##   Weave
1.  Two different startups : not the same description


In [14]:
df[df["Name"]== "Weave"]

Name  ...                                        Description
406  Weave  ...  Professional networking to arrange meetings an...
461  Weave  ...  Better phone service for your dental practice ...

[2 rows x 5 columns]

### Didn't drop any of them because we can get insights from the change of Class status

# Column 2: URLS

In [18]:
print(df["URL"].isna().sum())
df[df["URL"].isna() == True]

Name  ...                                        Description
29           Convrge  ...           Hang out with friends in virtual reality
33        FlightMate  ...    Flexible flight querying with automated booking
35   Jetty Financial  ...               Virtual assistant to manage finances
123        Redspread  ...          Version control for software environments
138      StrongIntro  ...  StrongIntro helps recruiters turn their entire...
..               ...  ...                                                ...
953            YouOS  ...  YouOS provides a platform for developers to cr...
954       ClickFacts  ...  ClickFacts is a Malware Software Solutions com...
955         Infogami  ...  Merged with Reddit.ÊInfogami is a wiki applica...
958           Memamp  ...                   A better desktop search solution
960          Simmery  ...                                                NaN

[153 rows x 5 columns]

## we have 153 startups without Url ==> 15.9%
Website is a redundant feature and will not give us any insight, we better drop the column

In [44]:
df.drop(columns=["URL"], axis=1, inplace=True)
df.head()

Name Class Status                                       Description
0       Awesound    F2    NaN                              AdWords for podcasts
1   BillionToOne    F2    NaN                    Fetal genetic testing in India
2       Bulletin    F2    NaN                                 Airbnb for retail
3        Cerplus    F2    NaN               B2B marketplace for surplus produce
4  ConnectHealth    F2    NaN  Telemedicine and prescription delivery in Africa

# Column 3: Class

In [20]:
print(df["Class"].isna().sum())

0


### Good thing, ,none of our startups has a missing status value

In [38]:
class_list=df["Class"].unique()
class_list.sort()
print(len(class_list) ,class_list )

24 ['F1' 'F2' 'S05' 'S06' 'S07' 'S08' 'S09' 'S10' 'S11' 'S12' 'S13' 'S14'
 'S15' 'W06' 'W07' 'W08' 'W09' 'W10' 'W11' 'W12' 'W13' 'W14' 'W15' 'W16']


### For all of our 962 startups we have only 24 class


# Column 4 : Status

In [39]:
df["Status"].isna().sum()

748

## 77.75% of our data is missing status

In [40]:
df["Status"].unique()

array([nan, 'Dead', 'Exited'], dtype=object)

In [42]:
dead_status= df[df["Status"]== 'Exited']
len(dead_status)

138

In [43]:
exited_status= df[df["Status"]== 'Exited']
len(exited_status)

76

From the remaining data : 

1.   76 starups have the exited status
2.   138 startups have the dead status



# Column 5 : Description

In [46]:
df.head()

Name Class Status                                       Description
0       Awesound    F2    NaN                              AdWords for podcasts
1   BillionToOne    F2    NaN                    Fetal genetic testing in India
2       Bulletin    F2    NaN                                 Airbnb for retail
3        Cerplus    F2    NaN               B2B marketplace for surplus produce
4  ConnectHealth    F2    NaN  Telemedicine and prescription delivery in Africa

In [47]:
df["Description"].isna().sum()

9

we only have 9 startups without description

In [49]:
startup_missing_descri= df[df["Description"].isna() == True]
startup_missing_descri

Name Class Status Description
778     I Move You   S10   Dead         NaN
788      Simperium   S10    NaN         NaN
819      Tweetflow   W10   Dead         NaN
832         Jobpic   S09   Dead         NaN
875     Precognate   S08   Dead         NaN
882  UrbanTakeover   S08   Dead         NaN
931          View3   W07   Dead         NaN
947      AudioBeta   W06    NaN         NaN
960        Simmery   S05   Dead         NaN

## Replacing nan values with -1 for model building

In [55]:
df.fillna(-1, inplace=True)

In [56]:
df

Name  ...                                        Description
0         Awesound  ...                               AdWords for podcasts
1     BillionToOne  ...                     Fetal genetic testing in India
2         Bulletin  ...                                  Airbnb for retail
3          Cerplus  ...                B2B marketplace for surplus produce
4    ConnectHealth  ...   Telemedicine and prescription delivery in Africa
..             ...  ...                                                ...
957          Loopt  ...  Acquired by Green Dot for $43.4MM.ÊLoopt is a ...
958         Memamp  ...                   A better desktop search solution
959         Reddit  ...  Acquired by Conde Nast in 2006.ÊA social news ...
960        Simmery  ...                                                 -1
961      TextPayMe  ...  Acquired by Amazon for ~$3MM.ÊAn SMS payment s...

[962 rows x 4 columns]

In [57]:
df.to_csv("y_combinator_data22")